### PR TITLE
fix(manhuagui): incorrect chapter order

### PR DESCRIPTION
**Pull request by dyphire**
Checklist:
- [x] Updated source's version for individual source changes
- [x] Updated all sources' versions for template changes
- [x] Set appropriate `nsfw` value
- [x] Did not change `id` even if a source's name or language were changed
- [x] Tested the modifications by running it on the simulator or a test device

### CODE REVIEW

File: src/rust/zh.manhuagui/src/decoder.rs [hunk 1]
```diff
@@ -36,8 +36,7 @@ impl Decoder {
 			String::new()
 		};
 
-		let suffix_vec = vec![
-			self.tr(c % self.a, 36),
+		let suffix_vec = [self.tr(c % self.a, 36),
```
Comment (Skittyblock) [inline src/rust/zh.manhuagui/src/decoder.rs:39]: why was the formatting here changed? it was better before

File: src/rust/zh.manhuagui/res/source.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
 		"id": "zh.manhuagui",
 		"lang": "zh",
 		"name": "ManHuaGui",
-		"version": 2,
+		"version": 3,
 		"url": "https://www.manhuagui.com/",
```
Comment (Skittyblock) [inline src/rust/zh.manhuagui/res/source.json:7]: should change this while we're at it
```suggestion
		"url": "https://www.manhuagui.com",
```

**Review by Skittyblock — CHANGES_REQUESTED** [review]
does this really change the chapter ordering? you're reversing the way you parse all the elements, but then you reverse the whole chapter list again, so doesn't it end up being the same? do you have an example manga that has incorrect ordering with the old version but is fixed with this version?

File: src/rust/zh.manhuagui/src/decoder.rs (hunk 1 above)
Comment (dyphire) [reply src/rust/zh.manhuagui/src/decoder.rs:39]: clippy would complain about it when testing in my repository, so fix that.

**Comment by dyphire** [comment]
> does this really change the chapter ordering? you're reversing the way you parse all the elements, but then you reverse the whole chapter list again, so doesn't it end up being the same? do you have an example manga that has incorrect ordering with the old version but is fixed with this version?

This is not true, it is done because of the special structure of the chapter list in this web page, which was overlooked by the previous people who implemented this source. 
It can be confirmed by checking the web page: https://www.manhuagui.com/comic/28004/

manhuagui v2: 
IMG_4931
This PR:
IMG_4933
IMG_4932

File: src/rust/zh.manhuagui/res/source.json (hunk 2 above)
Comment (dyphire) [reply src/rust/zh.manhuagui/res/source.json:7]: My computer has been turned off and can't be handled for the time being.

Comment (dyphire) [reply src/rust/zh.manhuagui/res/source.json]: done

**Comment by dyphire** [comment]
Any more questions here?

**Review by Skittyblock — APPROVED** [review]
alright, if you say it's correct then sure

**Pull request merged**